packages/nextjs/src/utils/isESM.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Determine if the given source code represents a file written using ES6 modules.
+ *
+ * The regexes used are from https://github.com/component/is-module, which got them from
+ * https://github.com/formatjs/js-module-formats, which says it got them from
+ * https://github.com/ModuleLoader/es-module-loader, though the originals are now nowhere to be found.
+ *
+ * @param moduleSource The source code of the module
+ * @returns True if the module contains ESM-patterned code, false otherwise.
+ */
+export function isESM(moduleSource: string): boolean {
```

Comment:
Should we add some basic tests for this function?

Reply:
+1

Also: Did we explore using the AST to figure out if a module is ESM or CJS?